examples/contract_template/README.md

```diff
@@ -0,0 +1,87 @@
+
+### Athena toolchain
+It's needed to build the contract code to the Risc-V RV32EM target. 
+This target is not yet natively supproted by the Rust toolchain. 
+We maintain a toolchain at https://github.com/athenavm/rustc-rv32e-toolchain.
+
+This repository contains a `cargo athena` plugin used to build the contracts at [cli](../../cli) as well as `athup` helper script to simplify installing it.
+
+To install `cargo athena` download and execute the athup:
+```sh
+curl -L https://raw.githubusercontent.com/athenavm/athena/main/athup/athup | bash
+```
+
+or execute it directly from within this repository at `athup/athup`.
+
+
+## Building the contract
+To build, simply execute the below command. The built executable is placed at `elf/contract_template`.
+
+```sh
+cargo athena build
+```
+
+## Testing the contract
+The template comes with some tests in [tests](tests/test.rs). They can be run with
+
+```sh
+cargo test
+```
+
+Mind there is no `athena` in the command because the tests are built and executed directly on your machine. They spin up the Athena VM and execute the smart contract code inside.
+
+## Deploying the contract onto the Athena test network
+
+A contract can be deployed using the standard Spacemesh single-sig wallet. It has a `deploy` method designed specifically for this purpose. 
+First, you will need a singlesig wallet account with some funds in it. One SMH is more than enough.
+
+Unfortunately, currently the Spacemesh wallet application doesn't support deploying contracts (yet). You will need to use the CLI tool developed specifically for the Athena test network. It's hosted at https://github.com/spacemeshos/go-spacemesh/tree/athena-poc/vm/cmd/client.
+
+It's written in Go, so you will need a working Go installation to proceed. Go can be installed by following the steps on https://go.dev/doc/install. Once it is installed, checkout the code at branch `athena-poc`:
+
+```sh
+git clone -b athena-poc git@github.com:spacemeshos/go-spacemesh.git
+
+```
+
+Next, go to `vm/cmd/client` directory. Create the single-sig wallet keys with
+```sh
+go run . generateKey
+```
+It will create a key.hex containing a private key for the wallet.
+
+Now we need to check the address of the wallet to move some funds to it. 
+```sh
+go run . coinbase
+```
+It will print the principal address of the wallet. You need to move funds to it or earn rewards for smeshing on the network as usual.
+
+The balance can be checked at the [explorer](https://explorer-devnet-athena.pages.dev/accounts). Once the account has some funds, it can be spawned:
+```sh
+go run . spawn --nonce 0 --address <address of public devnet node GRPC API>
+```
+
+Finally, it can be used to **deploy** the contract. Note: the nonce cannot be reused and should be increasing with each TX. The current counter value can also be checked on the explorer on the account page.
```

Comment:
```
Finally, it can be used to **deploy** the contract. 

> [!NOTE]
> The `nonce` cannot be reused and should be increasing with each TX. The current counter value can also be checked on the explorer on the account page.
```